src/worker.cpp

```diff
@@ -1267,6 +1267,17 @@ void Worker::collectFrusFromJson()
             continue;
         }
 
+#ifdef IBM_SYSTEM
+        // if chassis is powered on, skip collecting FRUs which are powerOffOnly
+        if (isChassisPowerOn() &&
+            jsonUtility::isFruPowerOffOnly(getSysCfgJsonObj(), vpdFilePath))
+        {
+            logging::logMessage("Skipping collection for FRU \'" + vpdFilePath +
```

Comment:
We don;t need a log for this else it will pollute journal on a good IPL as well.
I would suggest, club this condition with "if (vpdFilePath == SYSTEM_VPD_FILE_PATH)" and continue.

Reply:
Done